src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobWorker.cs

```diff
@@ -45,45 +57,71 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
 
             while (!cancellationToken.IsCancellationRequested)
             {
-                try
+                if (_searchParametersInitialized)
```

Comment:
Should there be a delay in this loop while it waits for this?

Reply:
Yes! That is a good idea.